Security/Firewall/GetJWTListener.php

```diff
@@ -98,17 +99,27 @@ public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
 
-        if ($this->options['post_only'] && !$request->isMethod('POST')) {
-            $event->setResponse(new JsonResponse('invalid method', 405));
-            return;
+        $is_json_request = $request->headers->get('content_type') === 'application/json';
+
+        if ($is_json_request) {
+            if (empty($request->getContent())) {
+                throw new \Exception('Error: Request parameters not found.');
+            }
         }
 
-        if ($this->options['post_only']) {
-            $username = trim($request->request->get($this->options['username_parameter'], null, true));
-            $password = $request->request->get($this->options['password_parameter'], null, true);
+        if (!$request->isMethod('GET')) {
```

Comment:
Let's explicitly check for POST and GET, because we should do nothing for other HTTP methods (but it shouldn't throw either).

```php
if (!in_array($request->getMethod(), ['POST', 'GET'])) {
    return;
}

if ($request->isMethod('POST')) {
```